src/main/java/com/jfrog/ide/common/persistency/XrayScanCache.java

```diff
@@ -19,17 +19,20 @@ public class XrayScanCache extends ScanCache {
      *
      * @param projectName - The IDE project name. If this is an npm project, it is a full path to the directory containing the package.json.
      * @param basePath    - The directory for the cache.
+     * @param quickScan   - Set to true to read cache from the file-system, if exist.
      * @param logger      - The logger.
      * @throws IOException in case of I/O problem in the paths.
      */
-    public XrayScanCache(String projectName, Path basePath, Log logger) throws IOException {
+    public XrayScanCache(String projectName, Path basePath, boolean quickScan, Log logger) throws IOException {
```

Comment:
I think that here the quickScan is a bit confusing. What do you think about something like `loadFromDisk` or `initNew`?